doc/rtd/topics/tutorials/qemu.rst

```diff
@@ -0,0 +1,282 @@
+``cd``.
+
+.. code-block:: sh
+
+   $ mkdir temp
+   $ cd temp
+
+
+Download a cloud image
+======================
+
+Cloud images typically come with cloud-init pre-installed and configured to run
+on first boot. Users should not need to worry about installing cloud-init
+unless they are manually creating their own images. In this case we select the
+latest Ubuntu LTS_.
+
+Download the server image using ``wget``.
+
+.. code-block:: sh
+
+    $ wget https://cloud-images.ubuntu.com/jammy/current/jammy-server-cloudimg-amd64.img
+
+
+Define our user data
+====================
+
+Create the following file ``user-data``. This user-data cloud-config
+sets the password of the default user and sets it to never expire. For
+more details see this module_.
+
+Execute the following command, which creates a file named ``user-data`` with
+configuration data.
+
+.. code-block:: sh
+
+    $ cat << EOF > user-data
+    #cloud-config
+    password: password
+    chpasswd:
+      expire: False
+
+    EOF
+
+
+What is user data?
+==================
+
+Before moving forward, let's inspect our user data file.
+
+.. code-block:: sh
+
+   $ cat user-data
+
+You should see the following contents:
+
+.. code-block:: yaml
+
+    #cloud-config
+    password: password
+    chpasswd:
+      expire: False
+
+The first line starts with ``#cloud-config``, which tells cloud-init
+what kind of configuration is contained. The cloud-config config type uses yaml
+format to tell cloud-init how to configure the virtual machine instance.
+Multiple different formats are supported by cloud-init. See the
+:ref:`documentation describing different formats<user_data_formats>`.
+
+The second line, ``password: password``, per :ref:`the docs<mod-users_groups>`,
+sets the default user's password to ``password``.
+
+The third and fourth lines direct cloud-init to set this default password to
+never expire.
+
+Define our meta data
+====================
+
+Execute the following command, which creates a file named ``meta-data`` with
+configuration data.
+
+.. code-block:: sh
+
+    $ cat << EOF > meta-data
+    instance-id: someid/somehostname
+    local-hostname: jammy
+
+    EOF
+
+
+Define our vendor data
+======================
+
+Now create the empty file ``vendor-data`` in your temporary directory. This
+will speed up the retry wait time.
+
+.. code-block:: sh
+
+    $ touch vendor-data
+
+
+Start an ad hoc IMDS Server
+===========================
+
+In a separate terminal, change to your temporary directory and then start the
+python webserver (built-in to python).
+
+.. code-block:: sh
+
+    $ cd temp
+    $ python3 -m http.server --directory .
+
+
+Launch a virtual machine with our user data
+===========================================
+
+Launch the virtual machine. By default, qemu will print to the terminal both
+kernel logs and systemd logs while the operating system boots. This may take a
+few moments to complete.
```

Comment:
```suggestion
Launch the virtual machine. By default, Qemu will print the kernel logs
and systemd logs to the terminal while the operating system boots. This
may take a few moments to complete.
```

Reply:
Just to make it completely crystal clear, I'd preface this little bit by saying something like "In our original terminal window" or something like that. Idiot that I am, I sat wondering which I was meant to use for a good minute until I realised that I couldn't use the IMDS window while the server was running.